core/storage/blog/gae_models.py

```diff
@@ -334,16 +334,26 @@ def has_reference_to_user_id(cls, user_id):
             cls.editor_ids == user_id).get(keys_only=True) is not None
 
     @classmethod
-    def get_by_user(cls, user_id):
-        """Retrieves the rights object for all blog posts assigned to given user
+    def get_multi_by_user(cls, user_id, published=False, limit=None):
+        """Retrieves the blog post rights objects for all blog posts for which
+        the given user is an editor.
 
         Args:
-            user_id: str. ID of user.
+            user_id: str. ID of the author of the blog post.
+            published: bool. Whether to fetch models for published blog posts.
```

Comment:
Given this definition, I think it's better to have get_all_published_models(user_id), get_all_draft_models(user_id), and get_multi_by_user(user_id, limit).

It's not clear that "published=False" means "draft" -- you're using a boolean to represent what should really be an enum. I think you're overloading this function too much (see the other comment about API design).

Reply:
Replaced this function with 3 functions: get_multi_published_models_by_user(user_id, limit) , get_multi_draft_models(user_id, limit), get_all_by_user(user_id).